fuzzers/030-iob/top.py

```diff
@@ -93,70 +115,151 @@ def run():
+
+            if p['type'] is None:
+                p['pad_wire'] = None
+            elif p['type'] == 'IBUF':
+                p['pad_wire'] = 'di[{}]'.format(i_idx)
+                p['IDELAY_ONLY'] = random.randint(0, 1)
+                if not p['IDELAY_ONLY']:
+                    p['owire'] = luts.get_next_input_net()
+                else:
+                    any_idelay = True
+                    p['owire'] = 'idelay_{site}'.format(**p)
+
                 p['DRIVE'] = None
-            p['SLEW'] = verilog.quote(random.choice(slews))
-            p['tristate_wire'] = random.choice(
-                ('0', luts.get_next_output_net()))
-            p['ibufdisable_wire'] = random.choice(
-                ('0', luts.get_next_output_net()))
-            p['intermdisable_wire'] = random.choice(
-                ('0', luts.get_next_output_net()))
-            io_idx += 1
-
-        if 'DRIVE' in p:
-            if p['DRIVE'] is not None:
-                p['DRIVE_STR'] = '.DRIVE({}),'.format(p['DRIVE'])
-            else:
-                p['DRIVE_STR'] = ''
+                p['SLEW'] = None
+                p['IBUF_LOW_PWR'] = random.randint(0, 1)
+
+                if iostandard in IN_TERM_ALLOWED:
+                    p['IN_TERM'] = random.choice(
+                        (
+                            'NONE',
+                            'UNTUNED_SPLIT_40',
+                            'UNTUNED_SPLIT_50',
+                            'UNTUNED_SPLIT_60',
+                        ))
+
+                i_idx += 1
+            elif p['type'] == 'IBUFDS':
+                p['pad_wire'] = 'di[{}]'.format(i_idx)
+                i_idx += 1
+                p['bpad_wire'] = 'di[{}]'.format(i_idx)
+                i_idx += 1
+
+                p['IDELAY_ONLY'] = random.randint(0, 1)
+                p['DIFF_TERM'] = random.randint(0, 1)
+                if not p['IDELAY_ONLY']:
+                    p['owire'] = luts.get_next_input_net()
+                else:
+                    any_idelay = True
+                    p['owire'] = 'idelay_{site}'.format(**p)
 
-        if p['type'] is not None:
-            tile_params.append(
-                (
-                    tile, site, p['pad_wire'], iostandard, p['DRIVE'],
-                    verilog.unquote(p['SLEW']) if p['SLEW'] else None,
-                    verilog.unquote(p['PULLTYPE'])))
-        params['tiles'].append(p)
+                p['DRIVE'] = None
+                p['SLEW'] = None
+                p['IBUF_LOW_PWR'] = random.randint(0, 1)
+
+            elif p['type'] == 'OBUF':
+                p['pad_wire'] = 'do[{}]'.format(o_idx)
+                p['iwire'] = luts.get_next_output_net()
+                if drives is not None:
+                    p['DRIVE'] = random.choice(drives)
+                else:
+                    p['DRIVE'] = None
+                p['SLEW'] = verilog.quote(random.choice(slews))
+
+                o_idx += 1
+            elif p['type'] == 'OBUFDS':
```

Comment:
Shouldn't there be also `OBUFT`? Maybe it is not needed, as there already is `OBUFTDS` providing the tristate information.